src/js/main.js

```diff
@@ -31,12 +31,13 @@ var testSuites = [];
 var testFilters = [];
 var currentTest;
 
+// Populate the device selection drop down menu after the gUM dialog closes.
 document.querySelector('gum-dialog').addEventListener('closed', function() {
-  if (typeof MediaStreamTrack.getSources === 'undefined') {
-    console.log('getSources is not supported, device selection not possible.');
-  } else {
-    MediaStreamTrack.getSources(gotSources);
-  }
+  navigator.mediaDevices.enumerateDevices()
+  .then(gotSources)
```

Comment:
I followed other examples [here](https://github.com/webrtc/samples/blob/master/src/content/getusermedia/source/js/main.js#L36)
and the example on [MDN](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Promise) at the bottom is the same.

But I can make it 4 if you prefer.